build_tools/cmake/iree_run_module_test.cmake

```diff
@@ -0,0 +1,222 @@
+    return()
+  endif()
+
+  set(_RUNNER_TARGET "iree-run-module")
+  iree_package_name(_PACKAGE_NAME)
+  set(_NAME "${_PACKAGE_NAME}_${_RULE_NAME}")
+
+  if(_RULE_XFAIL)
+    platform_xfail(_XFAIL ${_RULE_XFAIL})
+  endif()
+
+  if(_RULE_MODULE_SRC)
+    get_filename_component(_SRC "${_RULE_MODULE_SRC}" REALPATH)
+  elseif(_RULE_BENCHMARK_MODULE_SRC)
+    set(_MODULE_FLAG_DIR "${IREE_BENCHMARK_SUITE_DIR}/${_RULE_BENCHMARK_MODULE_SRC}/")
+    # Find the platform specific module flag file with matching path name.
+    # TODO(#10391): Update this logic with the new benchmark framework.
+    if(CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv64" AND
+       CMAKE_SYSTEM_NAME STREQUAL "Linux")
+      set(_FLAGFILE_HINT_PATH "${_MODULE_FLAG_DIR}/iree-llvm-cpu*RV64*__full-inference,default-flags/flagfile")
+    elseif(ANDROID)
+      set(_FLAGFILE_HINT_PATH "${_MODULE_FLAG_DIR}/iree-llvm-cpu*ARM64-v8A*__big-core,full-inference,default-flags/flagfile")
+    else()  # X86_64
+      set(_FLAGFILE_HINT_PATH "${_MODULE_FLAG_DIR}/iree-llvm-cpu*x86_64*__full-inference,default-flags/flagfile")
+    endif()
+    file(GLOB _FLAGFILE_PATH
+        LIST_DIRECTORIES FALSE
+        "${_FLAGFILE_HINT_PATH}"
+      )
+    if(_FLAGFILE_PATH)
+      get_filename_component(_FLAG_FILE_DIR "${_FLAGFILE_PATH}" DIRECTORY)
+      file(STRINGS "${_FLAGFILE_PATH}" _FLAGS ENCODING UTF-8)
+      # Parse the flagfile to find the vmfb location, entry_function,
+      # and function_input to append to the runner arguments.
+      foreach(_FLAG ${_FLAGS})
+        if(_FLAG MATCHES "--module_file=")
+          string(REPLACE "--module_file=" "" _SRC "${_FLAG}")
+          set(_SRC "${_FLAG_FILE_DIR}/${_SRC}")
+        elseif(_FLAG MATCHES "--device=")
+          string(REPLACE "--device=" "" _DEVICE "${_FLAG}")
+          # Override driver setting if it is set before.
+          set(_RULE_DRIVER "${_DEVICE}")
+        else()
+          list(APPEND _RULE_RUNNER_ARGS ${_FLAG})
+        endif()
+      endforeach(_FLAG)
+    elseif(NOT _FLAGFILE_PATH AND NOT _XFAIL)
+      message(SEND_ERROR "Cannot find the benchmark module for ${_RULE_BENCHMARK_MODULE_SRC}")
+    endif(_FLAGFILE_PATH)
+  endif(_RULE_MODULE_SRC)
+
+  list(APPEND _RULE_RUNNER_ARGS "--module_file=${_SRC}")
+
+  if(_RULE_EXPECTED_OUTPUT)
+    get_filename_component(_OUTPUT_FILE_TYPE "${_RULE_EXPECTED_OUTPUT}" EXT)
+    if(_OUTPUT_FILE_TYPE STREQUAL ".txt")
+      get_filename_component(_OUTPUT_FILE_SRC "${_RULE_EXPECTED_OUTPUT}" REALPATH)
+      # Process the text input to remove the line breaks.
+      file(STRINGS "${_OUTPUT_FILE_SRC}" _EXPECTED_OUTPUT ENCODING UTF-8)
+      string(REPLACE ";" " " _EXPECTED_OUTPUT_STR "${_EXPECTED_OUTPUT}")
+      set(_EXPECTED_OUTPUT_STR "--expected_output=\"${_EXPECTED_OUTPUT_STR}\"")
+      list(APPEND _RULE_RUNNER_ARGS ${_EXPECTED_OUTPUT_STR})
+      list(APPEND _SRC ${_OUTPUT_FILE_SRC})
+    elseif(_OUTPUT_FILE_TYPE STREQUAL ".npy")
+      find_file(_OUTPUT_FILE_SRC_PATH
```

Comment:
Why do we use a different method to find the file in this case?

Reply:
The npy file can be very big and does not exist in the codebase. `deeplab_v3_fp32_input_0_expected_output.npy` is a 6MB file and stored in GCS. I will add a comment to explain the logic better.